### PR TITLE
Feat/2062 group

### PR DESCRIPTION
**Pull request by benjamindehli**
## Description

Adds Summary2 support for groups and nested groups.

## Related Issue(s)

- closes #{issue number}

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [x] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/layout/Group/GroupSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+type GroupComponentSummaryProps = {
+  componentNode: LayoutNode<'Group'>;
+  childComponents: JSX.Element[] | null;
+  hierarchyLevel?: number;
+};
+
+export const GroupSummary = ({ componentNode, childComponents, hierarchyLevel = 0 }: GroupComponentSummaryProps) => {
+  const title = componentNode.item.textResourceBindings?.title;
+  const description = componentNode.item.textResourceBindings?.description;
+  const headingLevel = 3 + hierarchyLevel <= 6 ? 3 + hierarchyLevel : 6;
```
Comment (adamhaeger) [inline src/layout/Group/GroupSummary.tsx:18]: Vanskelig å lese logikken her, kanskje legge det inn i en funksjon?

Signaturen kan f eks være:

`const getLevel = (): HeadingProps['level'] => 1;`

Da bør du kunne slippe castingen på linje 24 under også.

File: src/layout/Group/index.tsx [hunk 2]
```diff
@@ -55,6 +55,10 @@ export class Group extends GroupDef implements ValidateComponent {
     );
   }
 
+  renderSummary2(): JSX.Element | null {
+    return <></>;
```
Comment (adamhaeger) [inline src/layout/Group/index.tsx:59]: Du må vel returnere noe her? Hvis ikke funker vel ikke Summary2?

File: src/layout/Group/GroupSummary.module.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.groupContainer.nested {
+  padding: var(--fds-spacing-6) var(--fds-spacing-4) var(--fds-spacing-6) var(--fds-spacing-4);
+  gap: var(--fds-spacing6);
```
Comment (adamhaeger) [inline src/layout/Group/GroupSummary.module.css:3]: tror det er skrivefeil i `--fds-spacing6`, skal vel være `--fds-spacing-6`?

**Review by adamhaeger — CHANGES_REQUESTED** [review]

File: src/layout/Group/GroupSummary.tsx (hunk 1 above)
Comment (benjamindehli) [reply src/layout/Group/GroupSummary.tsx:18]: Godt poeng 👍 Lager en funksjon for denne

File: src/layout/Group/GroupSummary.module.css (hunk 3 above)
Comment (benjamindehli) [reply src/layout/Group/GroupSummary.module.css:3]: Helt korrekt! Denne skal jeg få fikset

**Review by Magnusrm — APPROVED** [review]

**Review by adamhaeger — APPROVED** [review]

**Pull request merged**